pandas/core/resample.py

```diff
@@ -1028,7 +1028,7 @@ def __init__(self, freq='Min', closed=None, label=None, how='mean',
                  convention=None, base=0, **kwargs):
         freq = to_offset(freq)
 
-        end_types = set(['M', 'A', 'Q', 'BM', 'BA', 'BQ', 'W'])
+        self._end_types = end_types = {'M', 'A', 'Q', 'BM', 'BA', 'BQ', 'W'}
```

Comment:
huh?

Reply:
Ok, can see this is confusing. Moved it to class attribute.